src/pf-tooltip/pf-tooltip.component.js

```diff
@@ -0,0 +1,304 @@
+import {default as tmpl} from 'pf-tooltip.template';
+import {pfUtil} from 'pf-utils';
+
+/**
+ * <b>&lt;pf-tooltip&gt;</b> element for Patternfly Web Components
+ *
+ * @example {@lang xml}
+ * <pf-tooltip animation="fade" target="#btn-left" placement="left" delay="100" duration="150" container="#container"></pf-alert>
+ *
+ * @prop {string} animation the animation class
+ * @prop {string} target the target element selector
+ * @prop {string} placement left, right, top, bottom
+ * @prop {number} delay animation delay (ms)
+ * @prop {number} duration animation duration (ms)
+ * @prop {string} container the container element selector
+ */
+
+export class PfTooltip extends HTMLElement {
+
+  /**
+   * Reinitializes tooltip component with attribute values and resets content
+   */
+  init () {
+    this.element = this;
+    this.content = this.element.innerHTML;
+    this.tooltip = null;
+    this._target = this.getAttribute('target') ? document.querySelector(this.getAttribute('target')) : this;
```

Comment:
Yep..i'm fine with `targetSelector` ... i've seen `target` used in other frameworks, but that's fine with me. Will update it...